### PR TITLE
Setting sub_property on the entity

### PR DESCRIPTION
**Pull request by AronNovak**
#1030 

## Status
Addresses can be created with this patch, without that, the error in the issue appears.
image

### CODE REVIEW

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -698,7 +698,13 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missi
         $field_value = $this->propertyValuesPreprocess($property_name, $request[$public_field_name], $public_field_name);
       }
 
-      $wrapper->{$property_name}->set($field_value);
+      $sub_property = $info['sub_property'];
+      if (empty($sub_property)) {
+        $wrapper->{$property_name}->set($field_value);
+      }
+      else {
+        $wrapper->{$property_name}->{$sub_property}->set($field_value);
```
Comment (AronNovak) [inline plugins/restful/RestfulEntityBase.php]: @amitaibu is it correct or is it supposed to be handled on a higher level?

Comment (amitaibu) [reply plugins/restful/RestfulEntityBase.php]: I believe looks right.

**Comment by AronNovak** [comment]
@amitaibu should i create another issue to fix Travis there? Does it worth?

**Comment by amitaibu** [comment]
Yes, I'd be happy to see a working Travis.

**Comment by AronNovak** [comment]
@amitaibu now with green Travis

**Comment by AronNovak** [comment]
@amitaibu This is green, but I absolutely dislike the way how it's done, however at the project we need it, it does the job for the address field and it does not break tests there either.

**Pull request closed without merging**